samples/bullet-samples/02_shapes.py

```diff
@@ -0,0 +1,301 @@
+        base.cam.look_at(0, 0, 0)
+
+        # Light
+        alight = AmbientLight('ambientLight')
+        alight.set_color(LVector4(0.5, 0.5, 0.5, 1))
+        alightNP = render.attach_new_node(alight)
+
+        dlight = DirectionalLight('directionalLight')
+        dlight.set_direction(LVector3(1, 1, -1))
+        dlight.set_color(LVector4(0.7, 0.7, 0.7, 1))
+        dlightNP = render.attach_new_node(dlight)
+
+        render.clear_light()
+        render.set_light(alightNP)
+        render.set_light(dlightNP)
+
+        # Input
+        self.accept('escape', self.do_exit)
+        self.accept('r', self.do_reset)
+        self.accept('f1', base.toggle_wireframe)
+        self.accept('f2', base.toggle_texture)
+        self.accept('f3', self.toggle_debug)
+        self.accept('f5', self.do_screenshot)
+
+        inputState.watchWithModifiers('forward', 'w')
+        inputState.watchWithModifiers('left', 'a')
+        inputState.watchWithModifiers('reverse', 's')
+        inputState.watchWithModifiers('right', 'd')
+        inputState.watchWithModifiers('turnLeft', 'q')
+        inputState.watchWithModifiers('turnRight', 'e')
+
+        # Task
+        taskMgr.add(self.update, 'updateWorld')
+
+        # Physics
+        self.setup()
+
+    # _____HANDLER_____
+
+    def do_exit(self):
+        self.cleanup()
+        sys.exit(1)
+
+    def do_reset(self):
+        self.cleanup()
+        self.setup()
+
+    def toggle_debug(self):
+        if self.debugNP.is_hidden():
+          self.debugNP.show()
+        else:
+          self.debugNP.hide()
+
+    def do_screenshot(self):
+        base.screenshot('Bullet')
+
+    # ____TASK___
+
+    def process_input(self, dt):
+        force = LVector3(0, 0, 0)
+        torque = LVector3(0, 0, 0)
+
+        if inputState.isSet('forward'): force.y = 1.0
+        if inputState.isSet('reverse'): force.y = -1.0
+        if inputState.isSet('left'): force.x = -1.0
+        if inputState.isSet('right'): force.x = 1.0
+        if inputState.isSet('turnLeft'): torque.z = 1.0
+        if inputState.isSet('turnRight'): torque.z = -1.0
+
+        force *= 30.0
+        torque *= 10.0
+
+        self.boxNP.node().set_active(True)
+        self.boxNP.node().apply_central_force(force)
+        self.boxNP.node().apply_torque(torque)
+
+    def update(self, task):
+        dt = globalClock.get_dt()
+        self.process_input(dt)
+        self.world.do_physics(dt)
+        #pairs = [(mf.getNode0().getName(),
+        #          mf.getNode1().getName())
+        #  for mf in self.world.getManifolds() if mf.getNumManifoldPoints() > 0]
+        #print pairs
+        return task.cont
+
+    def cleanup(self):
+        self.world = None
+        self.worldNP.remove_node()
+
+        #def doAdded(self, node1, node2):
+        #  print 'added:', node1.getName(), node2.getName()
+
+        #def doDestroyed(self, node1, node2):
+        #  print 'destroyed:', node1.getName(), node2.getName()

```

Comment:
Either we should remove these two commented out functions if they have no use, or we should fix them to conform to the coding style (such as Python 3-compatible print functions).

Reply:
Since they're informative about the bullet API, I elect to keep and update them.